types/time.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import (
+	"sort"
+	"time"
+)
+
+func Now() time.Time {
+	return time.Now().Round(0).UTC()
+}
+
+func WeightedMedian(timeToVotingPower map[time.Time]int64) (res time.Time) {
+	totalVotingPower := int64(0)
+	for _, v := range timeToVotingPower {
+		totalVotingPower += v
+	}
+
+	median := totalVotingPower / 2
+
+	// To store the keys in slice in sorted order
+	var keys []time.Time
+	for k := range timeToVotingPower {
+		keys = append(keys, k)
```

Comment:
so maybe it would actually be better to have `WeightedMedian(weightedTimes []WeightedTime)` where 

```
type WeightedTime struct {
  Time time.Time
  Weight int64
}
```

Then we only have to build and sort the one array, instead of building up the map first and then copying into an array